src/main/java/Poirot.java

```diff
@@ -0,0 +1,204 @@
+        saveTasksToFile();
+    }
+
+    public static void saveTasksToFile() {
+        try {
+            File dir = new File("./data");
+            if (!dir.exists()) {
+                dir.mkdirs();
+            }
+            FileWriter writer = new FileWriter(FILE_PATH);
+            for (int i = 0; i < last_index; i++) {
+                writer.write(tasks[i].toFileString() + "\n");
+            }
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("Error saving tasks to file: " + e.getMessage());
+        }
+    }
+
+    public static void loadTasksFromFile() {
+        try {
+            File file = new File(FILE_PATH);
+            if (!file.exists()) {
+                return; // No file to load from
+            }
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                String[] parts = line.split(" \\| ");
+                String taskType = parts[0];
+                boolean isDone = parts[1].equals("1");
+                String description = parts[2];
+
+                switch (taskType) {
+                    case "T":
+                        add(new Todo(description));
+                        break;
+                    case "D":
+                        String by = parts[3];
+                        add(new Deadline(description, by));
+                        break;
+                    case "E":
+                        String from = parts[3];
+                        String to = parts[4];
+                        add(new Event(description, from, to));
+                        break;
+                    default:
+                        System.out.println("Corrupted data: " + line);
+                }
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("No saved tasks found.");
+        }
+    }
+
+    public static void markTask(int index, boolean isDone) {
+        tasks[index].setDone(isDone);
+        saveTasksToFile();
+    }
+    public static void deleteTask(int index) {
+        Task removedTask = tasks[index];
+        for (int i = index; i < last_index - 1; i++) {
+            tasks[i] = tasks[i + 1];
+        }
+        tasks[last_index - 1] = null;
+        last_index--;
+        System.out.println(LINE);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("  " + removedTask);
+        System.out.println("Now you have " + last_index + " tasks in the list.");
+        System.out.println(LINE);
+    }
+    public static void main(String[] args) {
+        System.out.println(LINE);
+        System.out.println("Hello! I'm POIROT\n");
+        System.out.println("What can I do for you?");
+        System.out.println(LINE);
+
+        loadTasksFromFile();
+
+        boolean working = true;
+        Scanner scan = new Scanner(System.in);
+        while (working) {
```

Comment:
Consider doing a saving changes to file at the end of each iteration, as such avoiding to call it every time something is changed, reduce duplicate lines